### ISSUE TITLE
Remix Cloudflare Pages - Your worker called response.clone()

### ISSUE THREAD

**Issue by shripadk** [remix-run/remix#6540]
### What version of Remix are you using?

Remix 1.16.1 and Wrangler version 3.0.1

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

### Steps to Reproduce

Setup project using [Remix Cloudflare Pages Template](https://github.com/remix-run/remix/tree/templates_v2_dev/templates/cloudflare-pages)

Then upgrade Wrangler package from 2.7.0 to 3.0.1

### Expected Behavior

Should not log any warnings to the console

### Actual Behavior

When using wrangler (powered by Miniflare and workerd), I get the following warning logged to the console when executing any actions:

```bash
Your worker called response.clone(), but did not read the body of both clones.
This is wasteful, as it forces the system to buffer the entire response body in 
memory, rather than streaming it through.

This may cause your worker to be unexpectedly terminated for going over
the memory limit. If you only meant to copy the response headers and
metadata (e.g. in order to be able to modify them), use 
`new Response(response.body, response)` instead.
```

Could track it down to this section of the code:

https://github.com/remix-run/remix/blob/880ab96ba02ac1d565028c9c8cbf9d98ca1edeb3/packages/remix-cloudflare-pages/worker.ts#L57-L60
Should be replaced with

```tsx
response = await context.env.ASSETS.fetch( 
 context.request.url, 
 new Response(response.body, response) 
); 
```

**Comment by pmbanugo** [comment]
The cause could also be because of the workers-sdk pages-asset-server. https://github.com/cloudflare/workers-sdk/blob/main/packages/pages-shared/asset-server/handler.ts#L340

**Comment by patdx** [comment]
I realized the suggested workaround above is passing a Response object to `ASSETS.fetch()`, which feels weird to me. I feel like we should strive to pass a Request object to `ASSETS.fetch()` instead. I think the Request object would do a better job of preserving important request info like the HTTP method, such as HEAD/GET/OPTIONS, which may affect how `ASSETS.fetch()` responds?

I've only tested in local wrangler dev mode so far, but the following seems to working for different incoming requests without showing the `request.clone()` error message:

```ts
const isGetOrHead = ['GET', 'HEAD'].includes(
  context.request.method.toUpperCase(),
);

const requestInit: RequestInit = isGetOrHead
  ? {}
  : {
      body: '',
    };

response = await context.env.ASSETS.fetch(
  new Request(context.request, requestInit),
);
```

Basically I am wrapping the original request with a new request that just has one difference. The original body is overridden with a fake body empty string `""`. I'm not exactly sure how `ASSETS.fetch()` handles POST requests and so on but I'm pretty sure it doesn't actually matter what the body is.

As defined in the [Request specification](https://fetch.spec.whatwg.org/#ref-for-dom-request%E2%91%A0):

> 35. If either init["body"] exists and is non-null or inputBody is non-null, and request’s method is `GET` or `HEAD`, then throw a TypeError.

This approach causes a TypeError when the request is GET or HEAD. So I made sure to check the method type before overriding the body or not.

Still testing it out but wanted to share the snippet in case it could be helpful!

**Comment by brophdawg11** [comment]
Thank you for opening this issue, and our apologies we haven't gotten around to it yet!  

With the release of [React Router v7](https://remix.run/blog/react-router-v7) we are sun-setting continued development/maintenance on Remix v2.  If you have not already [upgraded to React Router v7](https://reactrouter.com/upgrading/remix), we recommend you do so. We've tried to make the upgrade process as smooth as possible with our [Future Flags](https://remix.run/docs/en/main/start/future-flags). We are now in the process of cleaning up outdated issues and pull requests to improve the overall hygiene of our repositories.  

We plan to continue to address 2 types of issues in Remix v2:

* Bugs that pose security concerns
* Bugs that prevent upgrading to React Router v7

If you believe this issue meets one of those criteria, please respond or [create a new issue](https://github.com/remix-run/remix/issues/new?template=bug_report.yml).

For all other issues, ongoing maintenance will be happening in React Router v7, so:

* If this is a bug, please reopen this issue [in that repo](https://github.com/remix-run/react-router/issues/new?template=bug_report.yml) with a new [minimal reproduction](https://reactrouter.com/new) against v7
* If this is a feature request, please open a new [Proposal Discussion](https://github.com/remix-run/react-router/discussions/new?category=proposals) in React Router, and if it gets enough community support it can be considered for implementation

If you have any questions you can always reach out on [Discord](https://rmx.as/discord). Thanks again for providing feedback and helping us make our framework even better!
